src/Lean/Elab/Tactic/Simp.lean

```diff
@@ -265,7 +265,7 @@ def traceSimpCall (stx : Syntax) (usedSimps : UsedSimps) : MetaM Unit := do
     | .fvar fvarId => -- local hypotheses in the context
       if let some ldecl := lctx.find? fvarId then
         localsOrStar := localsOrStar.bind fun locals =>
-          if !ldecl.userName.isInaccessibleUserName &&
+          if !ldecl.userName.isInaccessibleUserName && !ldecl.userName.hasMacroScopes &&
```

Comment:
This seems like an unrelated change, does it affect the same test case? Should `isInaccessibleUserName` even be relevant here?

Reply:
I think it is indeed an unrelated bugfix. I believe this code is chosen to match `Name.toString`, since this is what is used to ultimately print the local names and it should be re-parseable, but I agree that it would have to be pretty weird code for the `isInaccessibleUserName` conjunct to trigger.

Reply:
Ok, if you could add a first test with a named hypothesis and then a second commit replacing `isInaccessibleUserName` with `hasMacroScopes` with the unnamed test case, that would be great.

Reply:
I.e. separating the context and inaccessible fix

Reply:
I've separated this into two commits, and verified both changes are needed.